python/cuml/test/test_arima.py

```diff
@@ -0,0 +1,572 @@
+    _, y = get_data()
+
+    mu = np.array([-217.7230173548441, -206.81064091237104])
+    ar = [np.array([0.0309380078339684]), np.array([-0.0371740508810001])]
+    ma = [np.array([-0.9995474311219695]), np.array([-0.9995645146854383])]
+
+    order = (1, 1, 1)
+
+    nb = 2
+
+    # model = arima.ARIMAModel(2*[order], mu, ar, ma, y)
+
+    x = arima.pack(1, 1, 1, nb, mu, ar, ma)
+
+    vs = arima._residual(nb, len(t), order, y, x)
+    # print("vs=", vs.T)
+
+    vs_ref = [[-3528.27698265, -115.83635122, 6935.3874952, -3813.00191784,
+               -1079.31878243, 6388.56331537, 2393.71237825, 1555.17359727,
+               1613.81554408, -680.33938589, -86.31011675, -2358.87207597,
+               3243.5007946, 1391.02259065, 403.4568625, 14.15395052,
+               -2745.78531069, 4259.68302997, 4704.57812106, 3433.65985293,
+               -12083.40867115, 4252.97752116, 3479.54897978, 6219.23655801,
+               2764.69001656, 3450.77699858, 9569.46467919, 5848.27045708,
+               -10014.66193498, -1123.21472226, -3054.02327057, -3394.91657265,
+               3786.62861209, 496.64301734, 5410.78600734, -7659.17583297,
+               2561.30086212, -2063.85623495, -3941.49054615, -2779.57418147,
+               -8595.08401264, 2955.05483003],
+              [-3614.18935909, -694.46090905, 4269.36176749, -3136.56635361,
+               -3050.69549741, 7478.21945144, 3266.35715354, 1028.87815306,
+               1739.43361009, -972.91523752, 835.37863546, -1553.38065136,
+               2587.8801488, 1560.66713759, 467.97432352, 266.04495099,
+               -1405.96052443, 3846.89140581, 4584.09935707, 4321.60001903,
+               -9832.32748628, 2411.48165741, 2378.86842679, 3160.94018602,
+               3644.24586102, 1449.94244736, 8839.76303398, 5294.62174579,
+               -8536.01598656, -1651.46330229, -2418.25003404, -3114.99846851,
+               4120.19930618, 711.01968571, 4356.34866645, -6769.92129656,
+               3375.88152766, -1718.57441537, -4424.02489802, -2652.16801305,
+               -8312.27984282, 3154.55704996]]
+
+    np.testing.assert_allclose(vs.T, vs_ref)
+
+def test_predict(plot=False):
+    _, y = get_data()
+
+    mu = [np.array([-217.7230173548441, -206.81064091237104]), np.array([-217.72325384510506, -206.77224439903458])]
+    ar = [[np.array([0.0309380078339684]), np.array([-0.0371740508810001])], [np.array([ 0.0309027562133337, -0.0191533926207157]), np.array([-0.0386322768036704, -0.0330133336831984])]]
+    ma = [[np.array([-0.9995474311219695]), np.array([-0.9995645146854383])], [np.array([-0.999629811305126]), np.array([-0.9997747315789454])]]
+
+    l2err_ref = [[7.611525998416604e+08, 7.008862739645946e+08], [7.663156224285843e+08, 6.993847054122686e+08]]
+
+    
+    for p in range(1, 3):
+        order = (p, 1, 1)
+
+        nb = 2
+
+        model = arima.ARIMAModel(2*[order], mu[p-1], ar[p-1], ma[p-1], y)
+
+        y_b_p = model.predict_in_sample()
+
+        if plot:
```

Comment:
I like to have the plotting option in the tests (with default false) because visualizing the results can be very helpful when dealing with tolerance problems. I can easily move the `matplotlib` to inside the if statement, so it never gets imported unless explicitly plotting.